sources/packages/web/src/components/aest/students/modals/EditStudentProfileModal.vue

```diff
@@ -0,0 +1,150 @@
+              type="date"
+              :max="getTodaysDate()"
+              hide-details="auto"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value
+            ><v-text-field
+              hide-details="auto"
+              label="Email"
+              v-model="formModel.email"
+              variant="outlined"
+              :rules="[checkEmailLengthRule, checkEmailValidationRule]"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+      </template>
+      <template #footer>
+        <check-permission-role :role="Role.StudentEditProfile">
+          <template #="{ notAllowed }">
+            <footer-buttons
+              :processing="loading"
+              primaryLabel="Update Profile"
+              @primaryClick="submit"
+              @secondaryClick="cancel"
+              :disablePrimaryButton="notAllowed"
+            />
+          </template>
+        </check-permission-role>
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, watchEffect, PropType } from "vue";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import TitleValue from "@/components/generic/TitleValue.vue";
+import { useFormatters, useModalDialog, useRules } from "@/composables";
+import {
+  UpdateStudentDetailsAPIInDTO,
+  UpdateStudentDetails,
+} from "@/services/http/dto";
+import { Role, VForm } from "@/types";
+
+export default defineComponent({
+  components: {
+    ModalDialogBase,
+    TitleValue,
+  },
+  props: {
+    updatedStudentDetails: {
+      type: Object as PropType<UpdateStudentDetails>,
+      required: true,
+    },
+  },
+  setup(props) {
+    const {
+      checkGivenNameLengthRule,
+      checkLastNameLengthRule,
+      checkEmailLengthRule,
+      checkEmailValidationRule,
+    } = useRules();
+    const { getISODateOnlyString } = useFormatters();
+    const getTodaysDate = (): string => {
+      return getISODateOnlyString(new Date());
+    };
+    const { showDialog, showModal, resolvePromise, loading, showParameter } =
+      useModalDialog<UpdateStudentDetailsAPIInDTO | boolean>();
+    const editStudentProfileForm = ref({} as VForm);
+    const formModel = ref({} as UpdateStudentDetailsAPIInDTO);
+    const submit = async () => {
+      const validationResult = await editStudentProfileForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      // Copying the payload, as reset makes the formModel properties null.
+      await resolvePromise(formModel.value);
+    };
+    watchEffect(() => {
+      formModel.value = {
+        givenNames: props.updatedStudentDetails.givenNames,
+        lastName: props.updatedStudentDetails.lastName,
+        birthdate: props.updatedStudentDetails.birthdate,
+        email: props.updatedStudentDetails.email,
+      };
+    });
```

Comment:
This works only for the first time the modal is loaded, which would not a blocker but if the values are provided as below it would allow eliminating this code, eliminating the prop and also ensure the values will be fresh every time modal is loaded.
```ts
const showModal = async (
  studentProfile: UpdateStudentDetails,
  canResolvePromise?: (
    value: UpdateStudentDetailsAPIInDTO | boolean,
  ) => Promise<boolean>,
) => {
  formModel.value = {
    givenNames: studentProfile.givenNames,
    lastName: studentProfile.lastName,
    birthdate: studentProfile.birthdate,
    email: studentProfile.email,
  };
  return showModalInternal(studentProfile, canResolvePromise);
};
```